Add: run gulp task based on UDP notifications server message

crysadrak opened the pull request:
Client side for usac link or another changes notification server

Commit:
Add: run gulp task based on UDP notifications server message

mjancarik commented on package.json:
```diff
@@ -57,6 +57,7 @@
     "browserify": "^16.2.2",
     "custom-event-polyfill": "^1.0.6",
     "del": "^3.0.0",
+    "dgram": "^1.0.1",
```
dgram module is deprectecated

mjancarik replied:
use dirrect https://nodejs.org/api/dgram.html

mjancarik commented on gulpConfig.js:
```diff
@@ -358,10 +368,17 @@ exports.files = {
 
 exports.occupiedPorts = {
   server: environment.$Server.port,
+  notifier: environment.notifier.port,
```
It is breaking change. It will end with throw error with port of undefined.

mjancarik commented on gulpConfig.js:
```diff
@@ -358,10 +368,17 @@ exports.files = {
 
+exports.notifyServerEnv = {
+  port: environment.notifier.port,
+  server: environment.notifier.server || 'localhost',
```
It is breaking change. It will end with throw error with server of undefined.

Commit:
Add: notification server disabled by default;
Add: queue jobs from notifications, timeout setup available from app environment;
Changed: task queue runs parallel instead of series

mjancarik commented on gulpConfig.js:
```diff
@@ -6,9 +6,15 @@ const fs = require('fs');
-  'less': ['less:build']
+const defaultNotifyServerEnv = {
+  enabled: false,
+  jobRunTimeout: 500,
```
Why is the default value set to 500?

Commit:
#! reduce notification server timeout, add readme
